packages/kbn-performance-testing-dataset-extractor/src/types.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { ToolingLog } from '@kbn/tooling-log';
+
+export interface Request {
+  transactionId: string;
+  spanId?: string;
+  name: string;
+  date: string;
+  duration: number;
+  http: {
+    method: string;
+    path: string;
+    params?: string;
+    body?: JSON | string;
+  };
+}
+
+export interface KibanaRequest extends Omit<Request, 'http'> {
```

Comment:
Do we absolutely need two different types here? If we need it, can we design the class hierarchy in a way that let's us avoid `Omit`? It might be due to my limited knowledge of Typescript but it strikes me as odd from a conceptual perspective that one would want to remove superclass properties in a subclass?